src/Longitude.cpp

```diff
@@ -83,12 +83,17 @@ const char *Longitude::formatString(char *targetBuffer, const char *format, long
     secs      = secs - degs * 3600;
     long mins = secs / 60;
     secs      = secs - mins * 60;
+    char sgn;
     if (totalSeconds < 0)
     {
-        degs = -degs;
```

Comment:
I'm not sure this will work correctly. `formatStringImpl` makes decisions based on the sign of `degs`, which would now behave differently.... we may need to look at all calls to `formatString` and see whether they pass `+`, which invokes that test.

Reply:
yep I think I got something wrong here.. when I checked the INDI logs I spotted 
Longitude 00:33:0.0 (0.55) (Longitude sign in carthography format) but this is not correct for my location .. the carthography longitude is -00:33:0 .. I am west of the prime meridian.  OAT does get the correct longitude it shows on the LCD display 0.54W but because I got this code wrong totalSeconds < 0 should be sgn = '+'...

formatStringImpl only makes the decision based on sign of degs if the char sgn = \0.. in my case I put a character so formatStringImpl simply copies sgn to the Meade command.

I did not spot this in earlier testing because I am so close to the meridian... I will do a bit more testing!

Reply:
Yep, I also did a grep through the code to look for were formatString is used.  For Longitude formatStringForMeade is used rather than the basic formatString code.  I agree that it is rather dangerous to assume that in the future someone might not see this and pass something else in with *format, however there is no protection in formatStringImpl for sgn and format '{+}' being used at the same time so you could get '++' generated by formatStringImpl anyway.  So to fix that I would make a change in formatStringImpl...